# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1273786601 14400
#      Thu May 13 17:36:41 2010 -0400
# Node ID 69a715c81eab5a1e36a68928f9a51eae971bef24
# Parent  10df75a24242bf5263d1df0277a87ad3e8950752
help.m: avoid unnecessary use of varargout

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2010-05-13  John W. Eaton  <jwe@octave.org>
+
+	* help/help.m: Avoid unnecessary use of varargout.
+
 2010-05-13   Qianqian Fang <fangq>
 
         * miscellaneous/tempdir.m: Add filesep to end of tempdir (bug #29864).
 
 2010-05-11  Jaroslav Hajek  <highegg@gmail.com>
 
 	* io/textread.m: Fix typo.
 
diff --git a/scripts/help/help.m b/scripts/help/help.m
--- a/scripts/help/help.m
+++ b/scripts/help/help.m
@@ -27,17 +27,17 @@
 ## Return the help text for the function, @var{name}.
 ## 
 ## The help command can give you information about operators, but not the
 ## comma and semicolons that are used as command separators.  To get help
 ## for those, you must type @kbd{help comma} or @kbd{help semicolon}.
 ## @seealso{doc, lookfor, which}
 ## @end deftypefn
 
-function varargout = help (name)
+function retval = help (name)
 
   if (nargin == 0)
 
     puts ("\n\
   For help with individual commands and functions type\n\
 \n\
     help NAME\n\
 \n\
@@ -78,17 +78,17 @@ function varargout = help (name)
     if (status != 0)
       warning ("help: Texinfo formatting filter exited abnormally; raw Texinfo source of help text follows...\n");
     endif
 
     if (nargout == 0)
       which (name);
       printf ("\n%s\n%s", text, __additional_help_message__ ());
     else
-      varargout = {text};
+      retval = text;
     endif
 
   else
     error ("help: invalid input\n");
   endif
 
 endfunction
 
